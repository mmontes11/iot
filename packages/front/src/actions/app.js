import {
  TOGGLE_HAMBURGER_MENU,
  TOGGLE_MAP_DIALOG,
  TOGGLE_LANGUAGE_SELECTOR,
  SET_SHOW_ERROR,
} from "constants/actionTypes/app";

export const toggleHamburgerMenu = () => dispatch => {
  dispatch({ type: TOGGLE_HAMBURGER_MENU });
};

export const toggleMapDialog = () => dispatch => {
  dispatch({ type: TOGGLE_MAP_DIALOG });
};

export const toggleLanguageSelector = () => dispatch => {
  dispatch({ type: TOGGLE_LANGUAGE_SELECTOR });
};

export const setShowError = showError => dispatch => {
  dispatch({ type: SET_SHOW_ERROR, showError });
};
