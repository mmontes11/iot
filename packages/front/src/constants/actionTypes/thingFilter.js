export const THING_FILTER_SELECT = "THING_FILTER_SELECT";
export const THING_FILTER_UPDATED = "THING_FILTER_UPDATED";
export const THING_FILTERS_REQUEST = "THING_FILTERS_REQUEST";
export const THING_FILTERS_REQUEST_SUCCESS = "THING_FILTERS_REQUEST_SUCCESS";
export const THING_FILTERS_REQUEST_ERROR = "THING_FILTERS_REQUEST_ERROR";
export const THING_FILTERS_UPDATED = "THING_FILTERS_UPDATED";
