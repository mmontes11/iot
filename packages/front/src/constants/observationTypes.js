export const EVENT_TYPE = "event";
export const MEASUREMENT_TYPE = "measurement";
export const OBSERVATION_TYPES = [EVENT_TYPE, MEASUREMENT_TYPE];
