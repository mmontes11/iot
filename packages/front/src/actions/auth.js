import {
  IS_AUTH,
  SET_USERNAME,
  SET_PASSWORD,
  LOGIN_REQUEST,
  LOGIN_REQUEST_SUCCESS,
  LOGIN_REQUEST_ERROR,
} from "constants/actionTypes/auth";
import { setShowError } from "actions/app";
import iotClient from "lib/iotClient";

export const isAuth = () => dispatch => {
  iotClient.authService
    .isAuth()
    .then(isAuthVal => dispatch({ type: IS_AUTH, isAuth: isAuthVal }))
    .catch(() => dispatch({ type: IS_AUTH, isAuth: false }));
};

export const setUsername = username => dispatch => {
  dispatch({ type: SET_USERNAME, username });
};

export const setPassword = password => dispatch => {
  dispatch({ type: SET_PASSWORD, password });
};

export const login = () => (dispatch, getState) => {
  dispatch({ type: LOGIN_REQUEST });
  const { username, password } = getState().auth;
  iotClient.authService.setCredentials(username, password);
  iotClient.authService
    .getToken()
    .then(response => {
      dispatch({
        type: LOGIN_REQUEST_SUCCESS,
        statusCode: response.statusCode,
        error: null,
      });
      isAuth()(dispatch);
    })
    .catch(error => {
      dispatch({
        type: LOGIN_REQUEST_ERROR,
        statusCode: error.statusCode,
        error,
      });
      setShowError(true)(dispatch);
    });
};

export const logout = () => dispatch => {
  iotClient.authService
    .logout()
    .then(() => {
      isAuth()(dispatch);
    })
    .catch(() => {
      isAuth()(dispatch);
    });
};
