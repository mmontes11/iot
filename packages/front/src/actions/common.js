import { RESET } from "constants/actionTypes/common";

export const reset = preserveError => dispatch => {
  dispatch({ type: RESET, preserveError });
};
