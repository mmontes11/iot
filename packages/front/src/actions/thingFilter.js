import {
  THING_FILTER_SELECT,
  THING_FILTER_UPDATED,
  THING_FILTERS_REQUEST,
  THING_FILTERS_REQUEST_SUCCESS,
  THING_FILTERS_REQUEST_ERROR,
  THING_FILTERS_UPDATED,
} from "constants/actionTypes/thingFilter";
import { ADD_FILTER_TYPE } from "constants/actionTypes/filters";
import iotClient from "lib/iotClient";
import { EVENT_TYPE, MEASUREMENT_TYPE } from "constants/observationTypes";
import { THING_FILTER_TYPE } from "constants/filterTypes";

export const selectThingFilter = (type, isActive) => dispatch => {
  dispatch({ type: THING_FILTER_SELECT });
  if (!isActive) {
    let getThingsParams = [];
    if (type === EVENT_TYPE) {
      getThingsParams = [undefined, true];
    } else if (type === MEASUREMENT_TYPE) {
      getThingsParams = [true, undefined];
    }
    dispatch({ type: THING_FILTERS_REQUEST });
    iotClient.thingsService
      .getThings(...getThingsParams)
      .then(res => {
        dispatch({ type: THING_FILTERS_REQUEST_SUCCESS });
        const things = res.body.things.map(thing => thing.name);
        dispatch({ type: THING_FILTERS_UPDATED, thingFilters: things });
      })
      .catch(error => {
        dispatch({ type: THING_FILTERS_REQUEST_ERROR, statusCode: error.statusCode, error });
      });
  }
};

export const updateThingFilter = thing => dispatch => {
  dispatch({ type: THING_FILTER_UPDATED, updatedThingFilter: thing });
};

export const addThingFilter = thing => dispatch => {
  dispatch({ type: ADD_FILTER_TYPE, addedFilterType: THING_FILTER_TYPE });
  dispatch({ type: THING_FILTER_UPDATED, updatedThingFilter: thing });
};
