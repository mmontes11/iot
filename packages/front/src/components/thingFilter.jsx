import React from "react";
import PropTypes from "prop-types";
import Dropdown from "components/dropdown";
import { THING_FILTER_TYPE } from "constants/filterTypes";
import { FormattedMessage } from "react-intl";

const ThingFilter = ({ thingFilter }) => (
  <div className="box">
    <div className="columns">
      <div className="column is-10">
        <Dropdown
          {...thingFilter}
          iconStyle="fa-angle-down"
          onButtonClick={() => thingFilter.onButtonClick()}
          onItemClick={item => thingFilter.onItemClick(item)}
        />
      </div>
      <div className="column is-center">
        <button
          id="delete-button"
          className="delete is-medium"
          onClick={() => thingFilter.onDelete(THING_FILTER_TYPE)}
          tabIndex={0}
        >
          <FormattedMessage id="Delete" />
        </button>
      </div>
    </div>
  </div>
);

ThingFilter.propTypes = {
  thingFilter: PropTypes.shape({
    label: PropTypes.string.isRequired,
    items: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    isActive: PropTypes.bool.isRequired,
    isLoading: PropTypes.bool.isRequired,
    isDisabled: PropTypes.bool.isRequired,
    onButtonClick: PropTypes.func.isRequired,
    onItemClick: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired,
  }).isRequired,
};

export default ThingFilter;
