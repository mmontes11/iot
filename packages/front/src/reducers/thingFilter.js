import {
  THING_FILTER_SELECT,
  THING_FILTER_UPDATED,
  THING_FILTERS_REQUEST,
  THING_FILTERS_REQUEST_SUCCESS,
  THING_FILTERS_REQUEST_ERROR,
  THING_FILTERS_UPDATED,
} from "constants/actionTypes/thingFilter";
import { TYPE_SELECT } from "constants/actionTypes/params";
import { RESET } from "constants/actionTypes/common";

export const initialState = {
  isLoading: false,
  isActive: false,
  isDisabled: false,
  items: [],
  selectedItem: null,
};

export default (state = initialState, { type, updatedThingFilter, thingFilters }) => {
  switch (type) {
    case TYPE_SELECT:
      return { ...state, isActive: false };
    case THING_FILTER_SELECT:
      return { ...state, isActive: !state.isActive };
    case THING_FILTER_UPDATED:
      return { ...state, selectedItem: updatedThingFilter, isActive: false };
    case THING_FILTERS_REQUEST:
      return { ...state, isLoading: true, isDisabled: true, isActive: false };
    case THING_FILTERS_REQUEST_SUCCESS:
      return { ...state, isLoading: false, isDisabled: false, isActive: false };
    case THING_FILTERS_REQUEST_ERROR:
      return { ...state, isLoading: false, isDisabled: false, isActive: false };
    case THING_FILTERS_UPDATED:
      return { ...state, items: thingFilters, isLoading: false, isDisabled: false, isActive: true };
    case RESET:
      return initialState;
    default:
      return state;
  }
};
