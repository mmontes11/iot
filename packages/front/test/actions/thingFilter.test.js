import { selectThingFilter, updateThingFilter } from "actions/thingFilter";
import { EVENT_TYPE, MEASUREMENT_TYPE } from "constants/observationTypes";

describe("actions/thingFilter", () => {
  it("dispatches a selectThingFilter action with all posible params", () => {
    let thunk = selectThingFilter(EVENT_TYPE, true);
    const dispatch = jest.fn();
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
    thunk = selectThingFilter(EVENT_TYPE, false);
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
    thunk = selectThingFilter(MEASUREMENT_TYPE, true);
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
    thunk = selectThingFilter(MEASUREMENT_TYPE, false);
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
  });
  it("dispatches a updateThingFilter action", () => {
    const thunk = updateThingFilter();
    const dispatch = jest.fn();
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
  });
});
