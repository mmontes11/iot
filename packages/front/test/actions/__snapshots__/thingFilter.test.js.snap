// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/thingFilter dispatches a selectThingFilter action with all posible params 1`] = `[Function]`;

exports[`actions/thingFilter dispatches a selectThingFilter action with all posible params 2`] = `[Function]`;

exports[`actions/thingFilter dispatches a selectThingFilter action with all posible params 3`] = `[Function]`;

exports[`actions/thingFilter dispatches a selectThingFilter action with all posible params 4`] = `[Function]`;

exports[`actions/thingFilter dispatches a updateThingFilter action 1`] = `[Function]`;
